packages/go_router/doc/configuration.md

```diff
@@ -0,0 +1,140 @@
+Create a GoRouter configuration by calling the [GoRouter][] constructor and
+providing list of [GoRoute][] objects:
+
+```dart
+GoRouter(
+  routes: [
+    GoRoute(
+      path: '/',
+      builder: (context, state) => const Page1Screen(),
+    ),
+    GoRoute(
+      path: '/page2',
+      builder: (context, state) => const Page2Screen(),
+    ),
+  ],
+);
+```
+
+# GoRoute
+To configure a GoRoute, a path template and builder must be provided. Specify a
+path template to handle by providing a `path` parameter, and a builder by
+providing either the `builder` or `pageBuilder` parameter:
+
+```dart
+GoRoute(
+  path: '/users/:userId',
+  builder: (context, state) => const UserScreen(),
+),
+```
+
+# Parameters
+To specify a path parameter, prefix a path segment with a `:` character,
+followed by a unique name, for example, `:userId`. You can access the value of
+the parameter by accessing it through the [GoRouterState][] object provided to
+the builder callback:
+
+```dart
+GoRoute(
+  path: '/users/:userId',
+  builder: (context, state) => const UserScreen(id: state.params['userId']),
+),
+```
+
+Similarly, to access a query parameter, use [GoRouterState][]:
+
+```dart
+GoRoute(
+  path: '/users',
+  builder: (context, state) => const UsersScreen(filter: state.queryParams['filter']),
+),
+```
+
+# Child routes
+A matched route can result in more than one screen being displayed on a
+Navigator. This is equivalent to calling `push()`, where a new screen is
+displayed above the previous screen with a transition animation, and with an
+in-app back button in the `AppBar` widget, if it is used.
+
+To display a screen on top of another, add a child route by adding it to the
+parent route's `routes` list:
+
+```dart
+GoRoute(
+  path: '/',
+  builder: (context, state) {
+    return HomeScreen();
+  },
+  routes: [
+    GoRoute(
+      path: 'details',
+      builder: (context, state) {
+        return DetailsScreen();
+      },
+    ),
+  ],
+)
+```
+
+# Nested navigation
+Some apps display destinations in a subsection of the screen, for example, an
+app using a BottomNavigationBar that stays on-screen when navigating between
+destinations.
+
+To add an additional Navigator, use
+[ShellRoute](https://pub.dev/documentation/go_router/ShellRoute-class.html)
+and provide a builder that returns a widget:
+
+```dart
+ShellRoute(
+  builder:
+      (BuildContext context, GoRouterState state, Widget child) {
+    return Scaffold(
+      body: child,
+      onTap: (idx) {
+          /* ... */
+      },
+      currentIndex: /* ... */,
+      bottomNavigationBar: BottomNavigationBar(
+        items: [
+          /* ... */
+        ],
+      ),
+    );
```

Comment:
I'm worried that the `Scaffold` boilerplate makes it difficult to focus on the important bit: the `Scaffold` wraps the `child` provided by the shell route's routes. What do you think of something like this?

```suggestion
    return Scaffold(
      body: child,
      /* ... */
      bottomNavigationBar: BottomNavigationBar(
        /* ... */
      ),
    );
```

Reply:
That makes sense, I'll change that.